server/patches/server/level/ServerPlayer.patch

```diff
@@ -17,11 +35,20 @@
              Logger logger = ServerPlayer.LOGGER;
  
              logger.getClass();
-@@ -1245,6 +1246,7 @@
+@@ -1245,6 +1257,7 @@
          this.canChatColor = serverboundclientinformationpacket.getChatColors();
          this.getEntityData().set(ServerPlayer.DATA_PLAYER_MODE_CUSTOMISATION, (byte) serverboundclientinformationpacket.getModelCustomisation());
          this.getEntityData().set(ServerPlayer.DATA_PLAYER_MAIN_HAND, (byte) (serverboundclientinformationpacket.getMainHand() == HumanoidArm.LEFT ? 0 : 1));
 +        this.viewDistance = java.util.OptionalInt.of(serverboundclientinformationpacket.viewDistance); // Loom
      }
  
      public ChatVisiblity getChatVisibility() {
+@@ -1331,7 +1344,7 @@
+ 
+     @Nullable
+     public Component getTabListDisplayName() {
+-        return null;
++        return tabListName;
```

Comment:
Once the above changes are made, the implementation can be referenced for this field. You can do something like `((org.loomdev.api.entity.player.Player) getLoomEntity()).getTabListName().orElse(null)`.

You'll have to move the text transformation here, however.